src/shared_modules/indexer_connector/tests/unit/monitoring_test.cpp

```diff
@@ -0,0 +1,105 @@
+/*
+ * Wazuh Indexer Connector - Monitoring tests
+ * Copyright (C) 2015, Wazuh Inc.
+ * September 08, 2023.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#include "monitoring_test.hpp"
+#include "monitoring.hpp"
+#include <chrono>
+#include <memory>
+#include <thread>
+
+/**
+ * @brief Test instantiation and check the availability of valid servers.
+ *
+ */
+TEST_F(MonitoringTest, TestInstantiationWithValidServers)
+{
+    const auto hostGreenServer {m_servers.at(0)};
+    const auto hostRedServer {m_servers.at(1)};
+
+    EXPECT_NO_THROW(m_monitoring = std::make_shared<Monitoring>(m_servers));
+
+    // All servers are available the first time.
+    EXPECT_TRUE(m_monitoring->isAvailable(hostGreenServer));
+    EXPECT_TRUE(m_monitoring->isAvailable(hostRedServer));
+
+    // Interval to check the health of the servers
+    std::this_thread::sleep_for(std::chrono::seconds(INTERVAL + 5));
+
+    // It's true because the green server is available
+    EXPECT_TRUE(m_monitoring->isAvailable(hostGreenServer));
+
+    // It's false because the red server isn't available
+    EXPECT_FALSE(m_monitoring->isAvailable(hostRedServer));
```

Comment:
Question: what do you think of not having `m_monitoring` as a member variable of the test class, and just declaring a Monitoring instance in each test case? It also looks as it does not need to be a pointer. (Unless I'm missing something).

Reply:
I decided to declare it as a shared pointer of the class to perform the EXPECT_NO_THROW on each test. If I declare it at the function level, this EXPECT cannot be done. I could move it as part of each function and remove the EXPECT.
should I move it as part of each function without the EXPECT_NO_THROW?

Reply:
Ok, I see. No problem.